doc/source/tutorial/interpolate/ND_regular_grid.rst

```diff
@@ -103,7 +103,61 @@ controlled by the ``fill_value`` keyword parameter:
     numerical artifacts. Consider rescaling the data before interpolating.
 
 
-Finally, we note that if you are dealing with data on Cartesian grids with
-integer coordinates, e.g. resampling image data, these routines may not be the
-optimal choice. Consider using `scipy.ndimage.map_coordinates` instead.
+.. _tutorial-interpolate_cartesian-grids:
+
+Uniformly spaced data
+=====================
+
+If you are dealing with data on Cartesian grids with integer coordinates, e.g.
+resampling image data, these routines may not be the optimal choice. Consider
+using `scipy.ndimage.map_coordinates` instead.
+
+For floating-point data on grids with equal spacing, ``map_coordinates`` can
+be easily wrapped into a `RegularGridInterpolator` look-alike. The following is
+a bare-bones example originating from `the Johanness Buchner's
+'regulargrid' package <https://github.com/JohannesBuchner/regulargrid/>`_::
+
+    class CartesianGridInterpolator:
+        def __init__(self, points, values, method='linear'):
+            self.limits = np.array([[min(x), max(x)] for x in points])
+            self.values = np.asarray(values, dtype=float)
+            self.order = {'linear': 1, 'cubic': 3, 'quntic': 5}[method]
+
+        def __call__(self, xi):
+            """
+            `xi` here is an array-like (an array or a list) of points.
+
+            Each "point" is an ndim-dimensional array_like, representing
+            the coordinates of a point in ndim-dimensional space.
+            """
+            # transpose the xi array into the ``map_coordinates`` convention
+            # which takes coordinates of a point along columns of a 2D array.
+            xi = np.asarray(xi).T
+
+            # convert from data coordinates to pixel coordinates
+            ns = self.values.shape
+            coords = [(n-1)*(val - lo) / (hi - lo)
```

Comment:
IMHO this doesn't handle reversed points dimensions, maybe just mention that the code relies on ascending points.

Reply:
OK, added a note in the last commit.